backend/alembic/versions/2023_06_05_1502-ffa8c33d129c_add_message_revisions.py

```diff
@@ -0,0 +1,54 @@
+"""add_message_revisions
+
+Revision ID: ffa8c33d129c
+Revises: 1b6e3ae16e9d
+Create Date: 2023-06-05 15:02:39.683151
+
+"""
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "ffa8c33d129c"
+down_revision = "1b6e3ae16e9d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "message_revision",
+        sa.Column("id", postgresql.UUID(as_uuid=True), nullable=False),
+        sa.Column("payload", postgresql.JSONB(astext_type=sa.Text()), nullable=True),
+        sa.Column("message_id", postgresql.UUID(as_uuid=True), nullable=False),
+        sa.Column("user_id", postgresql.UUID(as_uuid=True), nullable=True),
+        sa.Column(
+            "created_date", sa.DateTime(timezone=True), server_default=sa.text("CURRENT_TIMESTAMP"), nullable=True
+        ),
+        sa.ForeignKeyConstraint(
+            ["message_id"],
+            ["message.id"],
+        ),
+        sa.ForeignKeyConstraint(
+            ["user_id"],
+            ["user.id"],
+        ),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_index(op.f("ix_message_revision_message_id"), "message_revision", ["message_id"], unique=False)
+    op.add_column("message", sa.Column("edited", sa.Boolean(), server_default=sa.text("false"), nullable=False))
+    op.drop_index("idx_search_vector", table_name="message")
+    op.create_index("ix_search_vector", "message", ["search_vector"], unique=False, postgresql_using="gin")
+    # ### end Alembic commands ###
+
+
+def downgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index("ix_search_vector", table_name="message", postgresql_using="gin")
+    op.create_index("idx_search_vector", "message", ["search_vector"], unique=False)
```

Comment:
Is this re-creation of the message search index necessary?

Reply:
Ollie wrote via discord: "This is because the index in our DB is called "idx_search_vector" whereas in our code it's called "ix_search_vector". A problem from my message search implementation because I wrote the Alembic script manually instead of autogenerating it. It might be better to rename the one in the code to match the existing DB name and remove the Alembic create/drop"